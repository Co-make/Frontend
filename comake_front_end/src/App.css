.App {
  text-align: center;
}


.card {
  background: yellow;
  color: red;
}
