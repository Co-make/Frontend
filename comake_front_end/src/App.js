
import React, {useState} from 'react';
import List from './components/List';
import Register from './components/Register';
import Login from './components/Login';
import Profile from './components/Profile';
import AddIssue from './components/AddIssue';
import { Route, Redirect } from 'react-router-dom';
import {useLocalStorage} from './hooks/useLocalStorage';

function App() {

  const [token, setToken] = useLocalStorage('token', '')
  const [localId, setLocalId] = useLocalStorage('id', '')
  const [message, setMessage] = useState('')

  return (
    <div className="App">

      {/* ROUTES  */}

      <Route exact path="/" render={ props =>
          localStorage.getItem("token") ? (
            <List {...props} />
          ) : (
            <Redirect to="/login" />
          )
        }/>
      <Route exact path="/profile/:id" render={ props =>
          localStorage.getItem("token") ? (
            <Profile {...props} />
          ) : (
            <Redirect to="/login" />
          )
        } />
      <Route exact path="/login" render={ props =>
        <Login setToken={setToken}
        setLocalId={setLocalId}
        message={message}
        setMessage={setMessage}
        {...props} />}
        />

      <Route path="/register" render={ props =>
        <Register setToken={setToken}
        message={message}
        setMessage={setMessage}
        {...props}
      />}/>

      <Route path="/addIssue" render={ props =>
        <AddIssue setToken={setToken}
        message={message}
        setMessage={setMessage}
        {...props} />}
      />

    </div>
  );
}

export default App;
