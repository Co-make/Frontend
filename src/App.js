import React, { useState } from "react";
import List from "./components/List";
import Register from "./components/Register";
import Login from "./components/Login";
import Profile from "./components/Profile";
import AddIssue from "./components/AddIssue";
import { Route, Redirect } from "react-router-dom";
import { useLocalStorage } from "./hooks/useLocalStorage";
import Layout from "./components/Layout";

function App() {
  const [token, setToken] = useLocalStorage("token", null);
  const [localId, setLocalId] = useLocalStorage("id", "");
  const [zipCode, setZipCode] = useLocalStorage("zipcode", "");
  const [message, setMessage] = useState("");

  return (
    <div className="App">
<<<<<<< HEAD
      <header className="App-header">
        <nav className="nav-container">
          <Nav>
            <div className="logotext">
              <img className="logo" src={Logo}></img>omake
            </div>
            <a href="https://flamboyant-mayer-055230.netlify.com/aboutus.html">
              About
            </a>
            {token ? (
              <Link to="/login">
                <Button color="facebook" size="medium" onClick={logOut}>
                  Logout
                </Button>
              </Link>
            ) : token === "" ? (
              <></>
=======
      <Layout token={token} setToken={setToken} localId={localId}>
        <Route
          exact
          path="/"
          render={props =>
            localStorage.getItem("token") ? (
              <List {...props} />
>>>>>>> master
            ) : (
              <Redirect to="/login" />
            )
          }
        />
        <Route
          exact
          path="/profile"
          render={props =>
            localStorage.getItem("token") ? (
              <Profile {...props} />
            ) : (
              <Redirect to="/login" />
            )
          }
        />
        <Route
          exact
          path="/login"
          render={props => (
            <Login
              setToken={setToken}
              setLocalId={setLocalId}
              message={message}
              setMessage={setMessage}
              setZip={setZipCode}
              {...props}
            />
          )}
        />

        <Route
          path="/register"
          render={props => (
            <Register
              setToken={setToken}
              message={message}
              setMessage={setMessage}
              {...props}
            />
          )}
        />

        <Route
          path="/addIssue"
          render={props => (
            <AddIssue
              setToken={setToken}
              message={message}
              setMessage={setMessage}
              {...props}
            />
          )}
        />
      </Layout>
    </div>
  );
}

export default App;
