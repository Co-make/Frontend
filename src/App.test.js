import React from 'react';
import { BrowserRouter } from "react-router-dom";
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

afterEach(rtl.cleanup);

describe( 'Basic App functionality', ()=> {
  it('renders correctly', ()=> {
    const wrapper = rtl.render(<BrowserRouter><App /></BrowserRouter>)
    expect(wrapper.baseElement).toMatchSnapshot()
  })
  it('Log in renders', ()=> {
    const wrapper = rtl.render(
  <BrowserRouter><App /></BrowserRouter>
    );
    // console.log(wrapper.debug())
    const element = wrapper.queryByText(/Log in to your account/i);
    expect(element).toBeInTheDocument();
    expect(element).toBeTruthy();
    expect(element).toBeVisible();
  });
  // it('Profile button routes to user profile', ()=>{
  //   const wrapper = rtl.render(<BrowserRouter><App /></BrowserRouter>);
  //   const profileButton = wrapper.getByTestId('profile-button')
  //   // On main page, this is not the profile page
  //   expect(profilePageEditButton).not.toBeInTheDocument();
  //   // After clicking on button, I can see the edit profile button, therefore am on profile page
  //   rtl.fireEvent.click(profileButton);
  //   const profilePageEditButton = wrapper.queryByText(/Edit Profile/i)
  //   expect(profilePageEditButton).toBeInTheDocument()
  // })
})