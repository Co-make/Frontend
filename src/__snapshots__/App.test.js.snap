// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic App functionality renders correctly 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div>
        <nav
          class="sc-eNQAEJ haUAcN"
        >
          <a
            href="/"
          >
            <img
              class="sc-kGXeez jzXojl"
              src="CoMakeCircle.png"
            />
          </a>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 18h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zm0-5h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </nav>
        <div
          class="ui dimmer"
        >
          <div
            class="content"
          >
            <div
              class="ui text loader"
            >
              Loading
            </div>
          </div>
        </div>
        <div
          class="ui center aligned middle aligned grid"
          style="height: 100vh;"
        >
          <div
            class="column"
            style="max-width: 450px;"
          >
            <img
              class="ui medium centered image"
              src="sign-in-4.png"
            />
            <h2
              class="ui teal center aligned header"
            >
              Log in to your account
            </h2>
            <h4
              class="ui green center aligned header"
            />
            <h4
              class="ui red center aligned header"
            />
            <form
              class="ui large form"
            >
              <div
                class="ui stacked segment"
              >
                <div
                  class="field"
                >
                  <div
                    class="ui fluid left icon input"
                  >
                    <input
                      name="email"
                      placeholder="E-mail address"
                      type="text"
                      value=""
                    />
                    <i
                      aria-hidden="true"
                      class="user icon"
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="ui fluid left icon input"
                  >
                    <input
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                    <i
                      aria-hidden="true"
                      class="lock icon"
                    />
                  </div>
                </div>
                <button
                  class="ui facebook large fluid button"
                  type="submit"
                >
                  Log in
                </button>
              </div>
            </form>
            <div
              class="ui message"
            >
              New to us?
              <button
                class="ui mini positive button register-button"
              >
                Sign Up
              </button>
            </div>
          </div>
        </div>
        <footer
          class="bottom-nav sc-hMqMXs cGctzT"
        >
          <a
            class="sc-kpOJdX diMgIW"
            href="/addIssue"
          >
            <button
              aria-label="add"
              class="MuiButtonBase-root MuiFab-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiFab-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
          <div
            class="MuiButtonGroup-root"
            role="group"
          >
            <a
              class="MuiButtonGroup-grouped MuiButtonGroup-groupedText MuiButtonGroup-groupedText sc-ckVGcZ cvNdbu"
              color="default"
              href="/"
              variant="text"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              class="MuiButtonGroup-grouped MuiButtonGroup-groupedText MuiButtonGroup-groupedText sc-dxgOiQ AMShI"
              color="default"
              data-testid="profile-button"
              href="/profile"
              variant="text"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </footer>
      </div>
    </div>
  </div>
</body>
`;
